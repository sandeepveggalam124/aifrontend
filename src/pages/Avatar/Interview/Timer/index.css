.timer-container{
    text-align: center;
    display:flex
}

.clock-vector-class{
    width: 30%;
    height: 30%;
  
  }   
