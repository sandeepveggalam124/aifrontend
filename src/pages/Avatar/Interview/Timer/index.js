import React, { memo, useEffect,useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useTimer } from 'react-timer-hook';
import { ReactComponent as ClockVector } from './clockVector.svg';

import './index.css'

const Timer = memo(({ expiryTimestamp, displaySize, showCountdown,showClock,onExpireNavigate }) => {
  const[showClockVector,setShowClockVector]=useState(showClock)
  // useEffect{()=>{
  //   setShowClockVector(showClockVector)
  // }}
  const navigate = useNavigate();

  const {
    seconds,
    minutes,
    hours,
    days,
    isRunning,
    start,
  } = useTimer({ expiryTimestamp, onExpire: () => onExpireNavigate && navigate(onExpireNavigate) });

  useEffect(() => {
    start();
  }, [start]);

  return (
    <div className='d-flex flex-column justify-content-center align-items-center mt-2'>
    <div className='d-flex justify-content-center align-items-center'>
    {showClockVector &&
     <ClockVector className='clock-vector-class mx-1'/>
    }     
      <div style={{fontSize:displaySize}}>
        <span>{hours}</span>:<span>{minutes}</span>:<span>{seconds}</span>
      </div>
    </div>
    <div>
        <p>{isRunning && showCountdown ? `${minutes} minutes left to start the interview` : ''}</p>
    </div>
    </div>
  );
});

export default Timer;
