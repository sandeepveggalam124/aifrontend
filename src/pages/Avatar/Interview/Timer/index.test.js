import { MemoryRouter as Router } from "react-router-dom";
import {render} from "@testing-library/react";
import Timer from "./index";


it('renders correctly',()=>{
    const {container}=render(
        <Router>
            <Timer/>
        </Router>
    )
    expect(container).toMatchSnapshot()
})