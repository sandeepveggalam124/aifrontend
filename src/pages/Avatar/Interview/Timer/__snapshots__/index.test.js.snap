// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="d-flex flex-column justify-content-center align-items-center mt-2"
  >
    <div
      class="d-flex justify-content-center align-items-center"
    >
      <div>
        <span>
          0
        </span>
        :
        <span>
          0
        </span>
        :
        <span>
          0
        </span>
      </div>
    </div>
    <div>
      <p />
    </div>
  </div>
</div>
`;
